smart-contract/lottery/src/lottery.rs

```diff
@@ -244,88 +174,59 @@ impl Lottery {
     ///
     /// It performs the following actions:
     ///  - Adds the attached value to the prize pool, deducting the lottery fee.
-    ///  - Mints a new token representing the user's participation.
     ///  - Determines the outcome (jackpot, consolation prize, or no win).
     ///  - Distributes prizes and emits an event based on the outcome.
     #[odra(payable)]
     pub fn play_lottery(&mut self) {
-        self.assert_not_paused();
-        self.assert_active();
         self.assert_deposit();
-        let round_id = self.active_round.get_or_default();
-        let play_id = self.next_play_id.get_or_default();
+        let round_id = self.current_round.get_or_default();
+        let play_id = self.current_play_id.get_or_default();
         let caller = self.env().caller();
         self.collected_fees.add(self.lottery_fee.get_or_default());
         self.prize_pool
             .add(self.env().attached_value() - self.lottery_fee.get_or_default());
-        let meta = self.token_meta.get_or_default();
-        match self.cep47.mint(caller, vec![play_id], vec![meta]) {
-            Ok(_) => (),
-            Err(err) => self.env().revert(err),
-        }
-        self.next_play_id.add(U256::one());
+
+        let mut play_event = Play {
+            round_id,
+            player: caller,
+            play_id,
+            timestamp: self.env().get_block_time(),
+            is_jackpot: false,
+            prize: U512::zero(),
+        };
```

Comment:
That's not correct to use play event as a state holder. Let's track the local state variables (`prize` and `is_jackpot`) separately and later build the play even based on them.